src/cryptonote_core/service_node_list.cpp

```diff
@@ -1802,62 +1819,46 @@ namespace service_nodes
       return false;
     }
 
+    // NOTE: Deserialize long term state history
+    uint64_t bytes_loaded = 0;
     cryptonote::db_rtxn_guard txn_guard(m_db);
     std::string blob;
-    if (!m_db->get_service_node_data(blob))
+    if (m_db->get_service_node_data(blob, true /*long_term*/))
     {
-      return false;
-    }
-
-    std::stringstream ss;
-    ss << blob;
-    binary_archive<false> ba(ss);
+      bytes_loaded += blob.size();
+      LOKI_DEFER { blob.clear(); };
```

Comment:
Using a lambda + new temporary RAII object here seems like overkill versus just putting the `blob.clear();` by itself two lines later (after the `ss << blob;`).